SoftLayer/CLI/loadbal/order.py

```diff
@@ -41,8 +41,10 @@ def order(env, **args):
     """Creates a LB. Protocols supported are TCP, HTTP, and HTTPS."""
 
     mgr = SoftLayer.LoadBalancerManager(env.client)
+    network = SoftLayer.NetworkManager(env.client)
 
-    location = args.get('datacenter')
+    datacenter = network.get_datacenter(args.get('datacenter'))
+    location = {'id': datacenter[0]['id']}
```

Comment:
Add a check here to make sure we got a datacenter back, and produce an error if we didn't letting the user know the DC they want doesn't exist